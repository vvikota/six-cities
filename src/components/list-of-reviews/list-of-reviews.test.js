import React from 'react';
import renderer from 'react-test-renderer';
import ListOfReviews from './list-of-reviews.jsx';

const hotelCommentsMock = [
  {
    id: 1,
    user: {
      id: 1,
      isPro: true,
      name: `Alena`,
      avatarUrl: `url`,
    },
    rating: 10,
    comment: `text`,
    date: `april 2020`,
  }
];

it(`ListOfReviews is correctly renderer`, ()=> {
  const tree = renderer.create(<ListOfReviews
    hotelComments={hotelCommentsMock}
  />).toJSON();

  expect(tree).toMatchSnapshot();
});

