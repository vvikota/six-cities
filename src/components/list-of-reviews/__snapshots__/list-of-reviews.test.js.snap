// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListOfReviews is correctly renderer 1`] = `
Array [
  <h2
    className="reviews__title"
  >
    Reviews Â·
    <span
      className="reviews__amount"
    >
      1
    </span>
  </h2>,
  <ul
    className="reviews__list"
  >
    <li
      className="reviews__item"
    >
      <div
        className="reviews__user user"
      >
        <div
          className="reviews__avatar-wrapper user__avatar-wrapper"
        >
          <img
            alt="Reviews avatar"
            className="reviews__avatar user__avatar"
            height="54"
            src="url"
            width="54"
          />
        </div>
        <span
          className="reviews__user-name"
        >
          Alena
        </span>
      </div>
      <div
        className="reviews__info"
      >
        <div
          className="reviews__rating rating"
        >
          <div
            className="reviews__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "200%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <p
          className="reviews__text"
        >
          text
        </p>
        <time
          className="reviews__time"
          dateTime="april 2020"
        >
          Invalid Date
        </time>
      </div>
    </li>
  </ul>,
]
`;
